roles/host-provision/tasks/proxy.yml

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+- name: proxy | Set fact
+  set_fact:
+    proxy_ip: "{{ ansible_env['SSH_CLIENT'].split() | first }}"
+  # debug:
+  #   var: proxy_ip
+  tags: host-provision
+
+- name: proxy | Validate proxy connectivity
+  uri:
+    url: https://www.google.com
+    status_code: 200
+  environment:
+    https_proxy: "http://{{ proxy_ip }}:3128"
+  tags: host-provision
+
+- name: proxy | Add gateway instance to /etc/hosts
```

Comment:
The term "gateway-instance" comes from https://cloud.google.com/vpc/docs/special-configurations#proxyvm.  Given that, in this configuration, control node, proxy IP, and gateway instance are used interchangeably,  does it make sense to call it control-node instead?

Reply:
I have changed it to `control node` for all of the Ansible task names.
For the actual entries in config files, I have left it as `gateway-instance`, as it connects consistently to the main reference you mention (and I am referencing the link at the top of the `proxy.yml`).